src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -307,20 +311,24 @@ void UDPEndPointImplLwIP::HandleDataReceived(System::PacketBufferHandle && msg,
 
 CHIP_ERROR UDPEndPointImplLwIP::GetPCB(IPAddressType addrType)
 {
-    // IMPORTANT: This method MUST be called with the LwIP stack LOCKED!
```

Comment:
Was this controlling access to mUDP via the lock?  Put another way, can two racing calls to this now both enter the "mUDP == null" block in parallel?  Seems like they can.

Reply:
locking the lwip stack is done to protect lwip's internal state, i.e. in this case the target of `mUDP`. it incidentally also protects mUDP itself, but that was not the primary purpose of https://github.com/project-chip/connectedhomeip/pull/28798 where it was added. i am not familiar enough with CHIP to know if it should. can endpoint construction/destruction be racy? can we have a message being sent while an endpoing is being destroyed? i was under the impression that it CHIP serializes everything to its event loop task, so it's not racy in that way. is it?

Reply:
Matter itself should not be racy that way, as long as all accesses are on the Matter thread.

If that's meant to be a guarantee here, this function should `assertChipStackLockedByCurrentThread()`.